aws/resource_aws_elasticache_replication_group.go

```diff
@@ -304,6 +315,13 @@ func resourceAwsElasticacheReplicationGroupRead(d *schema.ResourceData, meta int
 
 	d.Set("replication_group_description", rgp.Description)
 	d.Set("number_cache_clusters", len(rgp.MemberClusters))
+	if aws.BoolValue(rgp.ClusterEnabled) {
+		if err := d.Set("cluster_mode", flattenElasticacheNodeGroupsToClusterMode(rgp.NodeGroups)); err != nil {
+			return fmt.Errorf("error setting cluster_mode attribute: %s", err)
+		}
+	} else {
+		d.Set("cluster_mode", []interface{}{})
+	}
```

Comment:
_minor_ this logic could live within the flatten function tbh?

Reply:
Its a little awkward because in order to keep backwards compatibility we have to use the other attribute. I can move it though.